pkgs/servers/sql/cockroachdb/generic.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, buildGoModule, fetchurl, cmake, xz, which, autoconf, ncurses6
+, libedit, libunwind, installShellFiles, removeReferencesTo, go, version, sha256
+}:
+
+buildGoModule rec {
+  pname = "cockroach";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
+    inherit sha256;
+  };
+  vendorSha256 = null;
+
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.cc.isGNU [
+    "-Wno-error=deprecated-copy"
+    "-Wno-error=redundant-move"
+    "-Wno-error=pessimizing-move"
+  ];
+
+  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
+  buildInputs = if stdenv.isDarwin then [ libunwind libedit ] else [ ncurses6 ];
+
+  postPatch = ''
+    patchShebangs .
+  '';
+  buildPhase = ''
+    runHook preBuild
+    make buildoss
```

Comment:
In https://github.com/NixOS/nixpkgs/pull/124378/files#r699984526 the recommendation was to use `makeFlags` and to overwrite the core detection logic in the makefile. Maybe it's not trivial, but at least not letting the makefile detect the building machine sounds like a good idea.

Reply:
`buildGoModule` doesn't respect `makeFlags` (or even run `make`) so it needs to done as it is currently.

Reply:
Ah, good to know. Maybe add a comment to the `buildPhase`.